Support dynamic select

Kuniwak opened the pull request:
This PR allows `select` to accept an array of channels. Previously, there was no way to `select` an array of channels. This PR only modifies functions related to `resultBuilders`, so the behavior of the channels will remain unchanged.

Kuniwak committed:
Support dynamic select

Kuniwak commented:
I found a mistake, so I am moving it back to draft.

Kuniwak committed:
Can take various sequences

Kuniwak commented:
I fixed the mistake.

gh123man commented:
Hey! Thanks for the contribution. This is a super interesting change and seems like a good addition to the library. 

I have a couple of suggestions: 

1. What do you think about calling `forEach` `any` instead? 

`forEach` implies an operation occurs for each element - however since select will choose only one operation per call, perhaps `any` would make more sense since select will chose "any one" channel based on it's availability. 

ex: 

```swift
 await select {
      any([a, b]) {
          receive($0) { await result <- $0! }
      }
  }
 ```

2. I think an overload to accept variadic parameters would compliment this change. ex:

```swift
public func any<T>(_ elements: T..., @SelectCollector cases: (T) -> ([SelectHandler])) -> [SelectHandler] {
    return elements.flatMap { cases($0) }
}
```

would allow us to also perform an operation like this

```swift
 await select {
    any(a, b) {
        receive($0) { await result <- $0! }
    }
}
```

let me know your thoughts.

Kuniwak committed:
Rename to `any` from `forEach`

Kuniwak committed:
Add an overload that can take variadic arguments

Kuniwak commented:
> What do you think about calling `forEach` `any` instead?

Sounds good. `any` is better, as you suggested. I renamed it at [2f00850](https://github.com/gh123man/Async-Channels/pull/7/commits/2f0085083626288034ba63bdc1b39bc8a29ee924).

> I think an overload to accept variadic parameters would complement this change

Also, I agreed. I added it at [9217527](https://github.com/gh123man/Async-Channels/pull/7/commits/9217527fc48f067b77ef6531a01800751d582be6).

gh123man commented:
Looks great. I'm going to merge this, do some additional testing, and publish this in the next release. Thanks again!
